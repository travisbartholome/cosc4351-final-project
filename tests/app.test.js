const request = require('supertest');
const express = require('express');

const app = require('../app');

describe('App', () => {
  it('should successfully render a page on GET to /', done => {
    request(app)
      .get('/')
      .expect(200, done);
  });
});

afterAll(() => {
  app.shutDown();
});
