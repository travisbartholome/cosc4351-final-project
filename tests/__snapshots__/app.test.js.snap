// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App views should render the cart page successfully 1`] = `
Object {
  "connection": "close",
  "content-length": "220",
  "content-type": "text/html; charset=utf-8",
  "etag": "W/\\"dc-cleHWUz5VZ3hXK92DHFx+Ux4+Ig\\"",
  "x-powered-by": "Express",
}
`;

exports[`App views should render the cart page successfully 2`] = `
"<!-- Placeholder -->
{&#34;products&#34;:[{&#34;id&#34;:4,&#34;price&#34;:6,&#34;description&#34;:&#34;Cheap product.&#34;,&#34;name&#34;:&#34;Product 4&#34;,&#34;image&#34;:&#34;product4.jpg&#34;}],&#34;total&#34;:6}
"
`;

exports[`App views should render the checkout page successfully 1`] = `
Object {
  "connection": "close",
  "content-length": "18",
  "content-type": "text/html; charset=utf-8",
  "etag": "W/\\"12-eMSL0FrAQMaqAs8n+jN4w6wC8Zo\\"",
  "x-powered-by": "Express",
}
`;

exports[`App views should render the checkout page successfully 2`] = `"<p>Total: 1200</p>"`;

exports[`App views should render the root/browse page successfully 1`] = `
Object {
  "connection": "close",
  "content-length": "6805",
  "content-type": "text/html; charset=utf-8",
  "etag": "W/\\"1a95-IfqoBpQQygAVPisr//eJuIltS/g\\"",
  "x-powered-by": "Express",
}
`;

exports[`App views should render the root/browse page successfully 2`] = `
"<!DOCTYPE html>
<html>
<!-- stock head file that contains all pages metadata-->
<head>
 <!-- stock head file that contains all pages metadata-->
<title>COSC4351 Bartholome-Le-Habib Ecommerce Store</title>
<meta charset=\\"utf-8\\">
<!--Automatically re-adjusts view based on device dimensions-->
<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
<!--Loading CSS from Bootstrap CDN-->
<link rel=\\"stylesheet\\" href=\\"https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css\\">
<script src=\\"https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js\\"></script>
<script src=\\"https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js\\"></script>
<link href=\\"https://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css\\" rel=\\"stylesheet\\">
 
 <link rel=\\"stylesheet\\" href=\\"/static/styles/browse.css\\" />
</head>
<body>
  <header>
    <!-- views/header.ejs -->
<!-- creates inverted (black) navigation bar w/ top that is fixed-->
<nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
<!-- \\"container-fluid => takes full width-->
<div class=\\"container-fluid\\">
    <div class=\\"navbar-header\\">
    <a class=\\"navbar-brand\\" href=\\"#\\"><span class=\\"glyphicon glyphicon-sunglasses\\"></span> Swag Store</a>
    </div>
    <ul class=\\"nav navbar-nav\\">
    <!-- Add in hyper link to potential Browse page-->
    <li class=\\"active\\"><a href=\\"/\\">Browse</a></li>
    <!-- Add in hyper link to Browse page -->
    <li><a href=\\"#\\">About Us</a></li>
    </ul>
    <ul class=\\"nav navbar-nav navbar-right\\">
    <!-- Add links to checkout and cart pages?-->
    <li><a href=\\"#\\"><span class=\\"glyphicon glyphicon-shopping-cart\\"></span> Cart</a></li>
    <li><a href=\\"/checkout\\"><span class=\\"glyphicon glyphicon-log-in\\"></span> Checkout</a></li>
    </ul>
    <form class=\\"navbar-form navbar-right\\" >
    <div class=\\"input-group\\">
        <input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Search\\" name=\\"search\\">
        <div class=\\"input-group-btn\\">
        <button class=\\"btn btn-default\\" type=\\"submit\\">
            <i class=\\"glyphicon glyphicon-search\\"></i>
        </button>
        </div>
    </div>
    </form>
</div>
</nav>
  </header>
  <div class=\\"content-container\\">
    <div class=\\"row\\">
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product1.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 1</h1>
            <p class=\\"price\\">$26.99</p>
            <p>A product.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product2.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 2</h1>
            <p class=\\"price\\">$9.99</p>
            <p>Another product.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product3.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 3</h1>
            <p class=\\"price\\">$10.50</p>
            <p>Wow, that&#39;s some product.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product4.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 4</h1>
            <p class=\\"price\\">$6.00</p>
            <p>Cheap product.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product5.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 5</h1>
            <p class=\\"price\\">$56.00</p>
            <p>Expensive product.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product6.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 6</h1>
            <p class=\\"price\\">$31.26</p>
            <p>Man we&#39;re kind of running out of these, huh?</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product7.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 7</h1>
            <p class=\\"price\\">$5.49</p>
            <p>Oh wow this is even cheaper.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product8.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 8</h1>
            <p class=\\"price\\">$10.69</p>
            <p>Nice.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product9.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 9</h1>
            <p class=\\"price\\">$0.99</p>
            <p>Dollar store shelf material v2.0.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
        <div class=\\"col-sm-4\\">
          <div class=\\"product-card\\">
            <div class=\\"img-container\\">
              <img class=\\"product-image\\" src=\\"/static/images/product10.jpg\\" alt=\\"Swag image\\" />
            </div>
            <h1>Product 10</h1>
            <p class=\\"price\\">$12.00</p>
            <p>Just your average product.</p>
            <p><button>Add to Cart</button></p>
          </div> 
        </div>
       
    </div>
  </div>
  <footer class=\\"container-fluid text-center\\">
    
<p>Â© 2020 Copyright: Le-Bartholome-Habib</p>
  

  </footer>
</body>
</html>
"
`;
